Catalog/Examples/RxSwift/Paging/RxPagingViewController.swift

```diff
@@ -0,0 +1,151 @@
+//
+//  RxPagingViewController.swift
+//  Catalog
+//
+//  Created by Filip Gulan on 13/09/2019.
+//  Copyright (c) 2019 Infinum. All rights reserved.
+//
+
+import UIKit
+import RxSwift
+import RxCocoa
+import Alamofire
+
+final class RxPagingViewController: UIViewController, Refreshable {
+
+    lazy var refreshControl: UIRefreshControl = {
+        let refreshControl = UIRefreshControl()
+        _tableView.refreshControl = refreshControl
+        return refreshControl
+    }()
+
+    // MARK: - Private properties
+
+    private let _disposeBag = DisposeBag()
+
+    private lazy var _dataSourceDelegate: TableDataSourceDelegate = {
+        return TableDataSourceDelegate(tableView: _tableView)
+    }()
+
+    // MARK: - IBOutlets
+
+    @IBOutlet private weak var _tableView: UITableView! {
+        didSet {
+            _tableView.tableFooterView = UIView(frame: .zero)
+            _tableView.registerClass(cellOfType: PokemonTableViewCell.self)
+            if #available(iOS 11.0, *) {
+                _tableView.contentInsetAdjustmentBehavior = .never
+            } else {
+                automaticallyAdjustsScrollViewInsets = false
+            }
+        }
+    }
+
+    // MARK: - Lifecycle -
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        _setupPagination()
+    }
+	
+}
+
+// MARK: - Extensions -
+
+private extension RxPagingViewController {
+
+
+    func _setupPagination() {
+        let item = UIBarButtonItem(title: "Sort", style: .plain, target: nil, action: nil)
+        navigationItem.rightBarButtonItem = item
+        
+        let sort = item.rx.tap
+            .asDriver()
+            .scan(false) { (state, _) in !state }
+        
+        let pullToRefresh = refreshControl.rx
+            .controlEvent(.valueChanged)
+            .asDriver()
+            .mapToVoid()
+
+        let willDisplayLastCell = _tableView.rx
+            .reachedBottomOnceWith(restart: pullToRefresh)
+
+        let pokemons = _pokemons(
+            loadNextPage: willDisplayLastCell,
+            reload: pullToRefresh.startWith(()), // Start initial load
```

Comment:
I'd move this `startWith` to the `_pokemons` function, just for logic separation.